### PR TITLE
Add XSLT transformation recipe

### PR DESCRIPTION
**Pull request by ammachado**
## What's changed?

Allow ChangePluginConfiguration to perform XSLT transformation on plugin configuration. Also, added a new recipe for arbitrary XSLT transformations.

## What's your motivation?

Perform XSLT transformations on XML files and on Maven plugin configuration.

## Anything in particular you'd like reviewers to focus on?

No

## Anyone you would like to review specifically?

No

## Have you considered any alternatives or workarounds?

FindAndReplace works for simple refactoring, but this allow more complex scenarios.

## Any additional context

No

### Checklist
- [x] I've added unit tests to cover both positive and negative cases
- [x] I've added the license header to any new files through `./gradlew licenseFormat`
- [x] I've used the IntelliJ IDEA auto-formatter on affected files

### CODE REVIEW

File: rewrite-maven/src/main/java/org/openrewrite/maven/ChangePluginConfiguration.java
Comment (timtebeek) [inline rewrite-maven/src/main/java/org/openrewrite/maven/ChangePluginConfiguration.java:103]: I'm wondering if we should simplify the logic here such that it's easier to follow and extend. Right now when there's both a configuration and xslt supplied, we don't do anything. That's perhaps confusing to folks. Instead I propose we let configuration to precedence, and update the logic to reflect that. Would you agree to that?
```suggestion
                        } else if (configuration != null) {
```

File: rewrite-maven/src/main/java/org/openrewrite/maven/ChangePluginConfiguration.java
Comment (timtebeek) [inline rewrite-maven/src/main/java/org/openrewrite/maven/ChangePluginConfiguration.java:107]:
```suggestion
                        } else if (xslt != null || xsltResource != null) {
```

File: rewrite-xml/build.gradle.kts
Comment (timtebeek) [inline rewrite-xml/build.gradle.kts:23]: We don't often log from our recipes, as it's hard to uniformly present that output in the various ways recipes might run. Instead we often opt to add markers, which we might then in turn log, or present in some other way to the user, as appropriate. If at all possible I'd opt for markers, or no logging statements here too. What are your thoughts on that?

File: rewrite-xml/src/main/java/org/openrewrite/xml/XsltTransformation.java
Comment (timtebeek) [inline rewrite-xml/src/main/java/org/openrewrite/xml/XsltTransformation.java:69]: Did you intend to add more of a description here? At first glance it seemed like that to me.

File: rewrite-xml/src/main/java/org/openrewrite/xml/XsltTransformation.java
Comment (timtebeek) [inline rewrite-xml/src/main/java/org/openrewrite/xml/XsltTransformation.java:60]: I'm wondering if it would make sense to add an xpath expression option to this recipe, such that we might not have to add the xslt options to `ChangePluginConfiguration`. This recipe can then match particular tags within specific files, and be used in isolation. Otherwise I might foresee XSLT options being added to more recipes, whereas that would complicate maintenance a bit going forward. Curious to hear how you'd feel about that.

File: rewrite-xml/src/main/java/org/openrewrite/xml/XsltTransformation.java
Comment (timtebeek) [inline rewrite-xml/src/main/java/org/openrewrite/xml/XsltTransformation.java:94]:
```suggestion
        return Preconditions.check(check, visitor);
```

File: rewrite-xml/src/main/java/org/openrewrite/xml/XsltTransformation.java
Comment (timtebeek) [inline rewrite-xml/src/main/java/org/openrewrite/xml/XsltTransformation.java:108]: Do the validated method calls here not imply the calls to `StringUtils.isBlank` ? I'd be quite surprised if we need to check these explicitly, but I'm not in my IDE as I go through this first review pass.

File: rewrite-xml/src/main/java/org/openrewrite/xml/XsltTransformationVisitor.java
Comment (timtebeek) [inline rewrite-xml/src/main/java/org/openrewrite/xml/XsltTransformationVisitor.java:76]:
```suggestion
```

File: rewrite-xml/src/main/java/org/openrewrite/xml/XsltTransformationVisitor.java
Comment (timtebeek) [inline rewrite-xml/src/main/java/org/openrewrite/xml/XsltTransformationVisitor.java:72]:
```suggestion
```

File: rewrite-xml/src/main/java/org/openrewrite/xml/XsltTransformationVisitor.java
Comment (timtebeek) [inline rewrite-xml/src/main/java/org/openrewrite/xml/XsltTransformationVisitor.java:38]:
```suggestion
```

**Review by timtebeek — COMMENTED** [review]
Impressive work already; thanks for getting this started! Curious to hear what you intend to use if for. I've gone through for a first round of review comments, let me know how you feel about what I've raised so far!

File: rewrite-xml/src/main/java/org/openrewrite/xml/XsltTransformationVisitor.java
Comment (ammachado) [reply rewrite-xml/src/main/java/org/openrewrite/xml/XsltTransformationVisitor.java:72]: Do you have an example on how can I use Markers instead of slf4j?  I'm using slf4j here to get this information when I run maven with execution debug output on (`mvn -X`).

File: rewrite-xml/src/main/java/org/openrewrite/xml/XsltTransformation.java
Comment (ammachado) [reply rewrite-xml/src/main/java/org/openrewrite/xml/XsltTransformation.java:108]: The idea here was to not let the user specify both `xslt` and `xsltResource` at the same time.

File: rewrite-maven/src/main/java/org/openrewrite/maven/ChangePluginConfiguration.java
Comment (ammachado) [reply rewrite-maven/src/main/java/org/openrewrite/maven/ChangePluginConfiguration.java:103]: Yes, that's a great idea. I had a clone of the ChangePluginConfiguration doing only the XSLT transformations on my branch.

File: rewrite-xml/build.gradle.kts
Comment (ammachado) [reply rewrite-xml/build.gradle.kts:23]: Logging is simpler, requires less knowledge of the intrinsics of OR.  But I'm interested in learning a different way to do this type of communication / user feedback. Is there any good example from where I can learn?

File: rewrite-xml/src/main/java/org/openrewrite/xml/XsltTransformation.java
Comment (ammachado) [reply rewrite-xml/src/main/java/org/openrewrite/xml/XsltTransformation.java:60]: The ChangePluginConfiguration, in my use case, its' the simplest transformation. The general case XsltTransfomation is being used in my setup to convert OSGI / Karaf blueprints to Spring Boot.  The XsltTransformation can do search & replace on XML files without converting them to PlainText files as wel..

File: rewrite-xml/build.gradle.kts
Comment (jkschneider) [reply rewrite-xml/build.gradle.kts:23]: Showing how markers are used. Agree with @timtebeek we cannot have logging statements in core recipes: https://app.moderne.io/recipes/org.openrewrite.java.search.FindMethods?organizationId=T3BlblJld3JpdGU%3D&defaults=W3sibmFtZSI6Im1ldGhvZFBhdHRlcm4iLCJ2YWx1ZSI6Im9yZy5vcGVucmV3cml0ZS5tYXJrZXIuU2VhcmNoUmVzdWx0IGZvdW5kKC4uKSJ9LHsibmFtZSI6Im1hdGNoT3ZlcnJpZGVzIiwidmFsdWUiOiJUcnVlIn1d

Comment (ammachado) [reply rewrite-xml/build.gradle.kts:23]: Your example uses DataTables which are not available on OpenRecipe yet, just on Moderne.io.

Comment (timtebeek) [reply rewrite-xml/build.gradle.kts]: Resolving this discussion as the logging was removed, and data tables are now available:
https://docs.openrewrite.org/running-recipes/data-tables

**Review by timtebeek — APPROVED** [review]
Great work! Added some minor polishing but otherwise good to go if you ask me. I'll leave a little room for further feedback before I merge, but will merge before the next release.

And thank for the kind contribution, explaining your intended use, and patience in seeing this through.

**Pull request merged**
